feat(curriculum): add intro to node js lessons (#63084)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/lecture-working-with-nodejs-and-event-driven-architecture/68ff6d39d7ed9ef36ff281f3.md

---
id: 68ff6d39d7ed9ef36ff281f3
title: How can you Install Node on your Computer?
challengeType: 19
dashedName: how-can-you-install-node-on-your-computer
---

# --description--

Now you’re ready to take your first steps into the world of Node.js by installing it on your computer. But first, before you install Node.js, we recommend installing a tool called NVM (Node Version Manager). As its name implies, NVM is a tool for managing multiple versions of Node.js on the same computer.

Setting up this tool from the start will help you in the future because you might need to switch between different versions of Node.js for different projects. With NVM, you can do this very easily.

> Let’s see how you can install NVM on macOS/Linux and Windows, and how you can use it to install Node.js.

Reviewer: ```suggestion
Let's see how you can install NVM on macOS/Linux and Windows, and how you can use it to install Node.js.
```

## How to Install NVM on macOS/Linux

On macOS and Linux, you can install NVM directly through a shell script.

But for macOS, note that the official documentation for NVM mentions that “you need to manually install the Xcode command line tools before running the install script, otherwise, it'll fail.”

> On macOS, you can install Xcode command line tools by running this command on your terminal:

Reviewer 2: ```suggestion
On macOS, you can install Xcode command line tools by running this command in your terminal:
```

```bash
xcode-select --version
```

Then, it’s time to install NVM. You can download and execute the NVM installation script by running one of these commands:

```bash
curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.40.3/install.sh | bash
```

```bash
wget -qO- https://raw.githubusercontent.com/nvm-sh/nvm/v0.40.3/install.sh | bash
```

To get the latest version of NVM at the time when you are reading this article, please refer to the official NVM documentation: `https://github.com/nvm-sh/nvm?tab=readme-ov-file#install--update-script`. There, you will find these commands with the latest version number.

You might need to restart your terminal when the process is completed to load the new command and the new configuration.

Then, run this command to confirm that NVM was installed correctly and to check your current version:

```bash
nvm -v
```

You should see your version of NVM in the output.

## How to install Node.js with NVM on macOS/Linux

Now that you have NVM installed on macOS/Linux, you can install Node.js.

To install the latest Long-Term Support (LTS) version of Node.js, run this command:

```bash
nvm install --lts
```

> Notice the `--lts` flag. This flag indicates that you want to install the current Long-Term Support (LTS) version of Node.js. This version is usually recommended because it prioritizes stability, reliability, and security over new or experimental features. It is thoroughly tested and ready for production. It’s also guaranteed to have longer support periods with a focus on bug fixes and security patches.

Reviewer: ```suggestion
Notice the `--lts` flag. This flag indicates that you want to install the current Long-Term Support (LTS) version of Node.js. This version is usually recommended because it prioritizes stability, reliability, and security over new or experimental features. It is thoroughly tested and ready for production. It's also guaranteed to have longer support periods with a focus on bug fixes and security patches.
```

Alternatively, you can install a specific version of Node.js by specifying the version number in the command:

```bash
nvm install <version>
```

For example, here’s how you can install Node.js v20:

```bash
nvm install 20
```

Once you finish installing a version of Node.js, you can switch to that version with this command:

```bash
nvm use <version>
```

For example, here’s how to switch to version 20:

```bash
nvm use 20
```

Finally, you can check the current version of Node.js with this command:

```bash
node -v
```

## How to install NVM on Windows

Now let’s see how you can install and use NVM on Windows.

Official versions of NVM only support Windows in specific cases. If you have Windows Subsystem for Linux (WSL) set up on your device, you can follow the macOS/Linux guide above to install and use NVM.

> However, for Windows itself there are a few alternatives. One of them is `nvm-windows`, a Node.js version management utility for Windows. This is a completely separate project than the official NVM project. But the purpose of these tools is the same — to help you work with multiple versions of Node.js on the same device.

Reviewer 2: ```suggestion
However, for Windows itself, there are a few alternatives. One of them is `nvm-windows`, a Node.js version management utility for Windows. This is a completely separate project from the official NVM project. But the purpose of these tools is the same — to help you work with multiple versions of Node.js on the same device.
```

```powershell
nvm -v
```

## How to Install Node.js using NVM for Windows

Now that you have `nvm-windows` installed on Windows, you are ready to install Node.js.

To install the latest Long-Term Support (LTS) version of Node.js, run this command:

```powershell
nvm install lts
```

The Long-Term Support (LTS) version is usually recommended, but you can install a different version by specifying its version number in the command:

```powershell
nvm install <version>
```

For example:

```powershell
nvm install 20
```

When the installation is completed, you can switch to that version of Node.js with this command:

```powershell
nvm use <version>
```

For example:

```powershell
nvm use 20
```

You can also check your current version of Node.js with this command:

```bash
node -v
```

## Basic Commands

Now that you know how to install NVM and Node.js, let’s check out some basic commands that you will find helpful when working with Node.js and NVM.

Also, while all these commands will work for NVM installed on macOS/Linux, including WSL, they may work differently or may not work at all for `nvm-windows`. We’ll leave alternative commands for `nvm-windows` whenever possible.

> ### How to Adjust Powershell’s Execution Policy

Reviewer: ```suggestion
### How to Adjust Powershell's Execution Policy
```
